homeassistant/components/redgtech/coordinator.py

```diff
@@ -0,0 +1,99 @@
+"""Coordinator for Redgtech integration."""
+
+from __future__ import annotations
+
+import logging
+from datetime import timedelta
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+from homeassistant.core import HomeAssistant
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ON, STATE_OFF, CONF_ACCESS_TOKEN, CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.exceptions import HomeAssistantError, ConfigEntryError, ConfigEntryNotReady
+from redgtech_api.api import RedgtechAPI, RedgtechAuthError, RedgtechConnectionError
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+@dataclass
+class RedgtechDevice:
+    """Representation of a Redgtech device."""
+    id: str
+    name: str
+    state: str
+
+
+type RedgtechConfigEntry = ConfigEntry[RedgtechDataUpdateCoordinator]
+
+
+class RedgtechDataUpdateCoordinator(DataUpdateCoordinator[list[RedgtechDevice]]):
+    """Coordinator to manage fetching data from the Redgtech API."""
+
+    config_entry: RedgtechConfigEntry
+
+    def __init__(self, hass: HomeAssistant, config_entry: RedgtechConfigEntry):
+        """Initialize the coordinator."""
+        self.api = RedgtechAPI()
+        self.access_token: str | None = None
+        self.email = config_entry.data[CONF_EMAIL]
+        self.password = config_entry.data[CONF_PASSWORD]
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(minutes=1),
+            config_entry=config_entry,
+        )
+
+    async def login(self, email: str, password: str) -> str:
+        """Login to the Redgtech API and return the access token."""
+        try:
+            access_token = await self.api.login(email, password)          
+        except Exception as e:
+            raise ConfigEntryError("Unexpected error during login") from e
+        self.access_token = access_token
+        return access_token
+
+    async def renew_token(self, email: str, password: str) -> None:
+        """Renew the access token."""
+        self.access_token = await self.api.login(email, password)
+        _LOGGER.debug("Access token renewed successfully")
+
+    async def _async_update_data(self) -> list[RedgtechDevice]:
+        """Fetch data from the API on demand."""
+        _LOGGER.debug("Fetching data from Redgtech API on demand")
+        try:
+            if not self.access_token:
+                self.access_token = await self.api.login(self.email, self.password)
+
+            data = await self.api.get_data(self.access_token)
+        except RedgtechAuthError:
+            _LOGGER.warning("Auth failed, trying to renew token")
```

Comment:
```suggestion
            _LOGGER.debug("Auth failed, trying to renew token")
```
I don't think that's worth a warning

Reply:
done